src/Shouldly/Internals/Is.cs

```diff
@@ -3,18 +3,18 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text.RegularExpressions;
-using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Shouldly
 {
     internal static class Is
     {
-        public static bool InRange<T>(T comparable, T @from, T to) where T : IComparable<T>
+        public static bool InRange<T>([DisallowNull] T comparable, [AllowNull] T @from, [AllowNull] T to) where T : IComparable<T>
```

Comment:
`[AllowNull]` and `[DisallowNull]` are the only way to annotate these parameters since C# still does not allow `T?` here. (Allowing `T?` without a class or struct constraint would be a CLR change.)